src/plots/plots.js

```diff
@@ -380,7 +380,7 @@ plots.supplyDefaults = function(gd) {
         if(_module.cleanData) _module.cleanData(newFullData);
     }
 
-    if(oldFullData.length === newData.length) {
+    if(oldFullData.length === newFullData.length) {
```

Comment:
How hard would it be to :lock: this fix?

Reply:
I was going to add a `Plotly.react` test in `transform_groupby_test` for this and [diffing on `_fullInput`](https://github.com/plotly/plotly.js/pull/2525#discussion_r179326587), hence https://github.com/plotly/plotly.js/pull/2525/commits/c2b11dc5a8665c8c1e86b7f7fc969b15184e86f2, since that's the obvious place where fullData will have a different length from data after finance is converted. But on second thought perhaps that can of worms should wait for #2508? I'll make a note in that issue if you 👍. The fact that all the existing `Plotly.react` and other tests pass means these didn't cause *problems* at least.

Reply:
> But on second thought perhaps that can of worms should wait for #2508?

sounds good :ok_hand: